client/src/components/Donation/wallets-button.tsx

```diff
@@ -52,51 +56,87 @@ const WalletsButton = ({
       disableWallets: ['browserCard']
     });
 
-    pr.on('token', event => {
-      const { token, payerEmail, payerName } = event;
-      setToken(token);
-      event.complete('success');
-      postPayment({
-        paymentProvider: PaymentProvider.Stripe,
-        token,
+    pr.on('paymentmethod', async event => {
+      const {
         payerEmail,
-        payerName
+        payerName,
+        paymentMethod: { id: paymentMethodId }
+      } = event;
+      //create payment intent
+      const {
+        data: { clientSecret, subscriptionId, error }
+      } = await createStripePaymentIntent({
+        email: payerEmail,
+        name: payerName,
+        amount,
+        duration
       });
-    });
 
-    void pr.canMakePayment().then(canMakePaymentRes => {
-      if (canMakePaymentRes) {
-        setPaymentRequest(pr);
-        checkPaymentPossibility(true);
-      } else {
-        checkPaymentPossibility(false);
+      if (error) {
+        event.complete('fail');
+        displayError(t('donate.try-another-method'));
+      } else if (clientSecret) {
+        // confirm payment intent
+        const { paymentIntent, error: confirmError } =
+          await stripe.confirmCardPayment(
+            clientSecret,
+            { payment_method: event.paymentMethod.id },
+            { handleActions: false }
+          );
+
+        if (confirmError) {
+          event.complete('fail');
+          displayError(t('donate.try-another-method'));
+        } else {
+          event.complete('success');
+          if (paymentIntent.status === 'requires_action') {
+            const { error } = await stripe.confirmCardPayment(clientSecret);
+            if (error) {
+              displayError(t('donate.try-another-method'));
+            } else {
+              postPayment({
+                paymentProvider: PaymentProvider.Stripe,
+                paymentMethodId,
+                payerEmail,
+                payerName,
+                subscriptionId
+              });
+            }
+          } else {
+            postPayment({
+              paymentProvider: PaymentProvider.Stripe,
+              paymentMethodId,
+              payerEmail,
+              subscriptionId
+            });
+          }
+        }
       }
     });
 
+    void pr.canMakePayment().then(result => {
+      if (result) setPaymentRequest(pr);
+    });
+
     return () => {
       setPaymentRequest(null);
-      checkPaymentPossibility(false);
     };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [label, amount, stripe, postPayment, handlePaymentButtonLoad]);
```

Comment:
It's tricky for me to test, but is there any reason _not_ to provide all the deps? You'd need to pull `displayError` inside the `useEffect` callback, but that seems fine.

Reply:
I added displayError accrodingly